lib/puppet/provider/service/rcng.rb

```diff
@@ -0,0 +1,48 @@
+Puppet::Type.type(:service).provide :rcng, :parent => :bsd do
+  desc <<-EOT
+    RCng service management with rc.d
+  EOT
+
+  defaultfor :operatingsystem => [:netbsd, :cargos]
+
+  def self.defpath
+    "/etc/rc.d"
+  end
+
+  # if the service file exists in rc.conf.d AND matches an expected pattern
+  # then it's already enabled
+  def enabled?
+    rcfile = File.join(rcconf_dir, @resource[:name])
+    if Puppet::FileSystem.exist?(rcfile)
+      File.open(rcfile).readlines.each do |line|
+        # Now look for something that looks like "service=${service:=YES}" or "service=YES"
+        if matchdata = line.match(/^#{@resource[:name]}=(?:YES|\${#{@resource[:name]}:=YES})/)
```

Comment:
As verified by putting junk into the rc file, it makes no difference to rcng if there is leading whitespace. So I've adjusted the regex in the provider too.